### DISCUSSION TITLE
How to display the frequency/deviation when a signal is decoded

### DISCUSSION THREAD

**Discussion by obones** [merbanan/rtl_433#2414] [Q&A] [answered]
Hello,

I have quite a few devices emitting around here and they are all decoded just fine by rtl_433 and my RTLSDR dongle.
As I'm trying to analyze the signals with other devices, I'd like to have a few more information about any given decode signal, in particular the center frequency (OOK/FSK) and the deviation (FSK).
Something like this output for instance
```
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2023-03-10 11:22:46
signal    : 433.85MHz
model     : TFA-Pool     Id        : 166
Channel   : 2            Battery   : 1             Temperature: 11.1 C       Integrity : CHECKSUM
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
time      : 2023-03-10 11:24:20
signal    : 868.12MHz +- 10kHz
model     : LaCrosse-TX35DTHIT                     id        : 61
Battery   : 1            NewBattery: 0             Temperature: 11.2 C       Humidity  : 60 %          Integrity : CRC
```

I looked a the command line options, but I could not find anything obvious in this area.

Is this something that is already available in memory and just a matter of adding to the output via new code?

**Comment by merbanan** [comment, marked as answer]
[-M time[:<options>] | protocol | level | noise[:<secs>] | stats | bits | help] Add various meta data to each output.

**Reply by obones** [reply]
Ah of course, the "level" option. It is in the general help but I thought it would only output the RSSI level.
Note that I would have preferred if frequencies were given with 3 digits but the current value is already giving me interesting details and I can always change it here and recompile my own version.
Do you think I should create a PR for this "3 digits frequency in metadata" change?

**Reply by merbanan** [reply]
I think the digits have been up for discussion before and the IIRC the tuner does not offer that good resolution. But maybe @zuckschwerdt  knows better.

**Reply by zuckschwerdt** [reply]
Indeed, 100 kHz estimates are the best we get here, based on the instant phase angle used. It's not an FFT ;)

**Reply by obones** [reply]
Well, here I see that the second digit is actually of importance here, especially because of the rounding that is done with only one digit. 
But I'll leave it as is and just keep a modified copy lying around whenever I need it.

**Reply by zuckschwerdt** [reply]
Expectation is the problem. People will see `433.85` and (rightfully) say the signal is at 433.85 MHz -- which it is most likely not.

**Reply by zuckschwerdt** [reply]
I already played with FFTW to good success in https://github.com/merbanan/rtl_433/tree/feat-beepdetect
It's planned to add a "true FFT" mode to get e.g. these stats at the expense of some cpu load.
